Description: This patch changes the default config paths
Author: Andreas Metzler <ametzler@debian.org>
Last-Update: 2010-08-07
Index: wmaker-crm/src/wconfig.h.in
===================================================================
--- wmaker-crm.orig/src/wconfig.h.in
+++ wmaker-crm/src/wconfig.h.in
@@ -116,7 +116,7 @@
  */
 
 /* list of paths to look for the config files, searched in order of appearance */
-#define DEF_CONFIG_PATHS "~/GNUstep/Library/WindowMaker:"PKGDATADIR
+#define DEF_CONFIG_PATHS "~/GNUstep/Library/WindowMaker:/etc/GNUstep/Defaults:"PKGDATADIR
 #define DEF_MENU_FILE	 "menu"
 
 /* name of the script to execute at startup */
