Description: This patch includes the debian paths
Author: Rodolfo García Peñas (kix) <kix@kix.es>
Last-Update: 2015-08-13
Index: WindowMaker-0.95.6/doc/wmaker.1
===================================================================
--- wmaker-crm.orig/doc/wmaker.in
+++ wmaker-crm/doc/wmaker.in
@@ -9,6 +9,9 @@ wmaker \- X11 window manager with a NEXT
 Window Maker is a X11 window manager with a NEXTSTEP look. It tries to
 emulate NeXT's look as much as possible, but it deviates from it as
 necessary.
+.PP
+\fIPlease note this manpage has been modified to reflect changes in the
+Debian installation of Window Maker.\fP
 .SH "OPTIONS"
 .TP
 .BI \-display " host:display.screen"
@@ -68,7 +71,7 @@ for a list of visuals available in your
 .PP
 .SH FILES
 .TP
-.B ~/GNUstep/Defaults/WindowMaker
+.B /etc/GNUstep/Defaults/
 general Window Maker defaults.
 .TP
 .B ~/GNUstep/Defaults/WMState
@@ -113,6 +116,9 @@ and here, in that order. Unless, the ind
 .B ~/GNUstep/Library/WindowMaker/Pixmaps/
 Window Maker looks for \fBpixmaps\fP here
 .TP
+.B ~/GNUstep/Library/Icons/
+Window Maker looks for \fBicons\fP here
+.TP
 .B ~/GNUstep/Library/WindowMaker/Backgrounds/
 Window Maker looks for backgrounds here
 .TP
