Description: Patch back libwraster symbol version to LIBWRASTER3.
Author: Andreas Metzler <ametzler@debian.org>
Bug-Debian: https://bugs.debian.org/811304
Origin: vendor

--- a/configure.ac
+++ b/configure.ac
@@ -76,6 +76,11 @@ WRASTER_REVISION=0
 WRASTER_AGE=0
 WRASTER_VERSION=$WRASTER_CURRENT:$WRASTER_REVISION:$WRASTER_AGE
 AC_SUBST(WRASTER_VERSION)
+dnl After next soname bump drop this and use WRASTER_VERSION as argument
+dnl for generate-mapfile-from-header.sh
+dnl http://bugs.debian.org/811304
+WRASTER_SYMBOL_VERSION=3
+AC_SUBST([WRASTER_SYMBOL_VERSION])
 dnl
 dnl libWINGs
 WINGS_CURRENT=4
--- a/wrlib/Makefile.am
+++ b/wrlib/Makefile.am
@@ -99,5 +99,5 @@ wrlib.pc: Makefile
 if HAVE_LD_VERSION_SCRIPT
 libwraster.map: $(include_HEADERS) $(top_srcdir)/script/generate-mapfile-from-header.sh
 	$(AM_V_GEN)$(top_srcdir)/script/generate-mapfile-from-header.sh \
-		-n LIBWRASTER -v $(WRASTER_VERSION) $(srcdir)/$(include_HEADERS) > libwraster.map
+		-n LIBWRASTER -v $(WRASTER_SYMBOL_VERSION) $(srcdir)/$(include_HEADERS) > libwraster.map
 endif
